modules/runner/src/main/ts/core/Utils.ts

```diff
@@ -0,0 +1,22 @@
+export const noop = (): void => {};
+
+export const makeUrl = (session: string, offset: number, failed: number, retry: number): string => {
+  const baseUrl = window.location.protocol + '//' + window.location.host + window.location.pathname;
+  if (offset > 0) {
+    const rt = (retry > 0 ? '&retry=' + retry : '');
+    return baseUrl + '?session=' + session + '&offset=' + offset + '&failed=' + failed + rt;
```

Comment:
Minor: `baseUrl` is not required for calculating the suffix / query parameters. So the function could just be `getQueryParams` which would return '' or this query parameter-based suffix above. Then the calculation of `baseUrl` could be done elsewhere. Not sure if you'd want that, though. Makes it a bit easier to test, because you wouldn't have window (or an equivalent object passed in).

Reply:
Done